File: listfragment/src/main/java/com/adazhdw/listfragment/utils/RecyclerUtil.java
package com.adazhdw.listfragment.utils;

import androidx.recyclerview.widget.RecyclerView;

public class RecyclerUtil {
    public static boolean isSlideToBottom(RecyclerView recyclerView) {
        if (recyclerView == null) return false;
        return recyclerView.computeVerticalScrollExtent() + recyclerView.computeVerticalScrollOffset()
                >= recyclerView.computeVerticalScrollRange();
    }
    public static boolean isAlreadyBottom(RecyclerView recyclerView) {
        if (recyclerView == null) return false;
        return !recyclerView.canScrollVertically(1);
    }
}

File: list-fragment/src/main/java/com/adazh/list/utils/RecyclerUtil.kt
package com.adazh.list.utils

import androidx.recyclerview.widget.RecyclerView

object RecyclerUtil {
    fun isSlideToBottom(recyclerView: RecyclerView?): Boolean {
        return if (recyclerView == null) false else recyclerView.computeVerticalScrollExtent() + recyclerView.computeVerticalScrollOffset() >= recyclerView.computeVerticalScrollRange()
    }

    fun isAlreadyBottom(recyclerView: RecyclerView?): Boolean {
        return if (recyclerView == null) false else !recyclerView.canScrollVertically(1)
    }
}
